spec/formatting.md

```diff
@@ -320,16 +319,20 @@ Implementation-defined _functions_ SHOULD use an implementation-defined _namespa
 
 #### Option Resolution
 
-The result of resolving _option_ values is an unordered mapping of string identifiers to values.
+The result of resolving _option_ values is an unordered mapping of string identifiers to values,
+determined as follows:
 
-For each _option_:
+1. Let `res` be a new empty mapping.
+1. For each _option_:
+   1. Let `id` be the string value of the _identifier_ of the _option_.
```

Comment:
Mention NFC here?

Should we check if `id` exists (which is an error)?

Should there be a step for initializing `res` with options found in the operand's resolved value (enabling option transitivity)?

Reply:
> Mention NFC here?

We should do that change separately, but I agree with it (as mentioned in https://github.com/unicode-org/message-format-wg/pull/885#pullrequestreview-2310712001).

> Should we check if `id` exists (which is an error)?

No, because that's a data model error and checked well before we get here.

> Should there be a step for initializing `res` with options found in the operand's resolved value (enabling option transitivity)?

No, that's something that a function handler needs to do internally. It needs to be able to separate the options that the operand carries and the options that are on the expression.

Reply:
> >     Should we check if id exists (which is an error)?
> No, because that's a data model error and checked well before we get here.

This feels like something that should be mentioned here?

> >    Should there be a step for initializing res with options found in the operand's resolved value (enabling option transitivity)?
> No, that's something that a function handler needs to do internally. It needs to be able to separate the options that the operand carries and the options that are on the expression.

This also feels like something that should be mentioned. This is a case where the expression might overwrite an option, btw.

Perhaps after the algorithm:

> [!NOTE]
> Implementations are required to check for duplicate _options_ 
> during parsing or when the _message_ is created from a data model representation.

> [!NOTE]
> _Options_ (and their values) that are part of the _resolved value_ of an _operand_
> might be added to the _resolved options_ by the _function handler_.

Reply:
> > > Should we check if id exists (which is an error)?
> > 
> > No, because that's a data model error and checked well before we get here.
> 
> This feels like something that should be mentioned here?

We have discussed this previously, e.g. in https://github.com/unicode-org/message-format-wg/pull/577#discussion_r1439577540, and I continue to disagree that we need a specific callout here about this data model error.

> > > Should there be a step for initializing res with options found in the operand's resolved value (enabling option transitivity)?
> >
> > No, that's something that a function handler needs to do internally. It needs to be able to separate the options that the operand carries and the options that are on the expression.
> 
> This also feels like something that should be mentioned. This is a case where the expression might overwrite an option, btw.

This section is specifically about resolving the _option_ values on the current _expression_. I'm a bit concerned that adding a note about something that's ultimately internal to _function handlers_ seems like it would add confusion.

Reply:
You're adding an algorithm here that was previously just some prose. Algorithms are also more fussy, since you have to specify the corner cases and such. A naive implementer would read this and just pick up the options specified in the expression--missing the fact that the operand might be bringing some to the party--especially because the first step is _Let `res` be a new empty mapping_ and does not contain a step "import the options from the resolved value (if any)"

I didn't want to make this more complicated than it is, because it's fine. But why not add the note about function handler behavior?

We don't need to include the other note if you feel strongly about it.

Reply:
Added the note about function handler behavior.